backend/api/Services/MissionService.cs

```diff
@@ -346,5 +354,59 @@ MissionQueryStringParameters parameters
             // Constructing the resulting lambda expression by combining parameter and body
             return Expression.Lambda<Func<Mission, bool>>(body, mission);
         }
+
+        private static void ApplySort(ref IQueryable<Mission> missions, string orderByQueryString)
+        {
+            if (!missions.Any())
+                return;
+
+            if (string.IsNullOrWhiteSpace(orderByQueryString))
+            {
+                missions = missions.OrderBy(x => x.Name);
+                return;
+            }
+
+            string[] orderParams = orderByQueryString
+                .Trim()
+                .Split(',')
+                .Select(t => t.Trim())
+                .ToArray();
+            var propertyInfos = typeof(Mission).GetProperties(
+                BindingFlags.Public | BindingFlags.Instance
+            );
+            var orderQueryBuilder = new StringBuilder();
+            foreach (string param in orderParams)
+            {
+                if (string.IsNullOrWhiteSpace(param))
+                    continue;
+                string propertyFromQueryName = param.Split(" ")[0];
+                var objectProperty = propertyInfos.FirstOrDefault(
+                    pi =>
+                        pi.Name.Equals(
+                            propertyFromQueryName,
+                            StringComparison.InvariantCultureIgnoreCase
+                        )
+                );
+                if (objectProperty == null)
+                    throw new InvalidDataException(
+                        $"Mission has no property '{propertyFromQueryName}' for ordering"
+                    );
+                string sortingOrder = param.EndsWith(" desc", StringComparison.OrdinalIgnoreCase)
+                  ? "descending"
+                  : "ascending";
+#pragma warning disable CA1305
+                orderQueryBuilder.Append($"{objectProperty.Name} {sortingOrder}, ");
+#pragma warning restore CA1305
+            }
+
+            string orderQuery = orderQueryBuilder.ToString().TrimEnd(',', ' ');
+            if (string.IsNullOrWhiteSpace(orderQuery))
+            {
+                missions = missions.OrderBy(x => x.Name);
```

Comment:
Can we choose a more descriptive name than `x`?

Reply:
Yes!